tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/remove_participant_data_past_deactivation_date_test.py

```diff
@@ -0,0 +1,143 @@
+from cdr_cleaner.cleaning_rules.remove_participant_data_past_deactivation_date import (
+    RemoveParticipantDataPastDeactivationDate)
+from constants.retraction.retract_deactivated_pids import DEACTIVATED_PARTICIPANTS
+from tests.integration_tests.data_steward.cdr_cleaner.cleaning_rules.bigquery_tests_base import BaseTest
+
+
+class RemoveParticipantDataPastDeactivationDateTest(
+        BaseTest.CleaningRulesTestBase):
+
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+        super().initialize_class_vars()
+
+        # set the test project identifier
+        project_id = os.environ.get(PROJECT_ID)
+        cls.project_id = project_id
+
+        # set the expected test datasets
+        dataset_id = os.environ.get('COMBINED_DATASET_ID')
+        cls.dataset_id = dataset_id
+        sandbox_id = f"{dataset_id}_sandbox"
+        cls.sandbox_id = sandbox_id
+
+        cls.kwargs = {
+            'table_namer': 'bar_ds',
+            'api_project_id': 'foo-project-id'
+        }
+        cls.rule_instance = RemoveParticipantDataPastDeactivationDate(
+            project_id, dataset_id, sandbox_id, **cls.kwargs)
+
+        sb_table_names = cls.rule_instance.get_sandbox_tablenames()
+        cls.fq_sandbox_table_names = [
+            f'{project_id}.{sandbox_id}.{table_name}'
+            for table_name in sb_table_names
+        ]
+
+        # append table name here to ensure proper cleanup
+        cls.fq_sandbox_table_names.append(
+            f"{project_id}.{sandbox_id}.{DEACTIVATED_PARTICIPANTS}")
+
+        cls.fq_table_names = [
+            f"{project_id}.{dataset_id}.{tablename}"
+            for tablename in cls.rule_instance.affected_tables
+        ]
+
+        cls.fq_obs_table = [
+            table for table in cls.fq_table_names if 'observation' in table
+        ][0]
+
+        # call super to set up the client, create datasets, and create
+        # empty test tables
+        # NOTE:  does not create empty sandbox tables.
+        super().setUpClass()
+
+    def setUp(self):
+        """
+        Add data to the tables for the rule to run on.
+        """
+        insert_fake_data_tmpls = [
+            self.jinja_env.from_string("""
+        INSERT INTO `{{fq_table_name}}`
+        (observation_id, person_id, observation_concept_id, observation_date,
+         observation_type_concept_id, observation_source_concept_id)
+        VALUES
+        -- Values to exist after running the cleaning rule --
+          (801, 1, 1585899, date('2019-05-01'), 45905771, 111111),
+          (802, 2, 1585899, date('2019-05-01'), 45905771, 222222),
+        -- Values that should be removed by the cleaning rule --
+          (804, 1, 1585899, date('2020-05-01'), 45905771, null),
+          (805, 3, 1585899, date('2020-05-01'), 45905771, 45)
```

Comment:
Thank you, I think these are helpful. I think it would also be helpful to indicate the reasons. E.g. 801 is retained because it is before 1's deactivation, 2 is not deactivated, 804 is after 1 deactivated and 805 is after 3 deactivated.